### ISSUE TITLE
GPE killing Eclipse autocomplete

### ISSUE THREAD

**Issue by GoogleCodeExporter** [tsaiyilien/google-plugin-for-eclipse#49]
```
Whenever I try to use autocomplete feature in Java code I get error:

"com.google.appengine.eclipse.core.java.JdbcUrlCompletor' proposal from the 
'com.google.appengine.eclipse.core' plug-in did not complete normally. The 
extension violated the API contract of the 'computeContextInformation()' 
operation.
(see the attached screenshot)

I can reproduce this on two systems I use: Eclipse 3.6 (on Ubuntu 9.04) and 
Eclipse 3.7 (on Mint 12). 

My Eclipse Indigo installation is quite fresh and clean - after installing bare 
Eclipse I installed GEP for AppEngine development and required dependencies. I 
also have Lombok (http://projectlombok.org/) installed for my eclipse 
workspaces, not sure if that could somehow break GPE.


```

Original issue reported on code.google.com by `radomi...@gmail.com` on 23 Jan 2012 at 5:24

Attachments:
- [broken-proposal.png](https://storage.googleapis.com/google-code-attachments/google-plugin-for-eclipse/issue-49/comment-0/broken-proposal.png)

**Comment by GoogleCodeExporter** [comment]
```
I was having this problem in Ubuntu as well, using an old copy of Galileo that 
went haywire on this.  When I downloaded and installed a fresh copy of Indigo, 
the problem went away.  When I had the problem, I did not have lombok.org 
installed, but I did have a number of other plugins, like android and maven 
installed.

Also note, in an attempt to make the plugin work, I noticed it was based on 
JDBC, so I installed support for all the data tools I could find from the repo. 
 Make sure you have all the Data Tools installed.  Also, try to download a new 
copy from eclipse, unzip into a new folder, and run it from there.

If you are unable to fix this, try posting your eclipse run / config scripts so 
we can compare any differences we have.
```

Original comment by `jamie.ne...@promevo.com` on 23 Jan 2012 at 7:17

**Comment by GoogleCodeExporter** [comment]
```
I have the same Error.

Here is the list of plugins intalled on my Eclipse.


1.  Eclipse
2.  D: \ Eclipse \ AOSI-EclipseD2I-v1.5.2
3.  Apache MyFaces Trinidad Tag Support (Optional) 2.2.0.v20090517-20-7w311A1636
4.  Axis2 Tools 1.1.0.v200905242110-78-FBpDZRDE6FdPdLYYe
5.  D21 Maven 1.5.0
6.  Graphical Editing Framework Zest Visualization Toolkit 
1.1.1.v20090724-1146-66088D5DH8G6N4GNJ53563
7.  Graphical Editing Framework Draw2d 3.5.1.v20090724-1146-4407w3118192402452
8.  Maven: The Definitive Guide book (Optional) 0.9.8.200905041414
9.  Maven Doxia Editors (Optional) 0.9.8.200905041414
10. Maven Embedder 2.1.0.20080530-2300
11. Maven Integration for Eclipse (Required) 0.9.8.200905041414
12. Maven Integration for WTP (Optional) 0.9.8.200905041414
13. Maven issue tracking configurator for Mylyn (Optional) 0.9.5.20080717-1821
14. Maven issue tracking configurator for Mylyn 3.x (Optional) 
0.9.8.200905041414
15. Maven POM Editor (Optional) 0.9.8.200905041414
16. Maven POM XML Editor (Optional) 0.9.8.200905041414
17. Maven SCM handler for Team COS (Optional) 0.9.8.200905041414
18. Maven SCM Integration (Optional) 0.9.8.200905041414
19. D21 SON 1.5.0
20. JavaHL 1.6.0 Win32 Binaries (Optional) 2.2.1.120091009-1900
21. Native JavaHL 1.6 Implementation (Optional) 2.2.1.120091009-1900
22. Subversive SON Connectors 2.2.1.120091009-1900
23. Subversive SON Connectors 2.2.1.120091009-1900
24. Subversive SON Integration for the Mylyn Project (Optional) (Incubation) 
0.7.8.120091023-1300
25. Subversive SON JDT Ignore Extensions (Optional) (Incubation) 
0.7.8.120091023-1300
26. Subversive SON Team Provider (Incubation) 0.7.8.120091023-1300
27. Subversive SON Team Provider UI Capabilities (Incubation) 
0.7.8.120091023-1300
28. SVNKit 1.1.7 Implementation (Optional) 2.2.1.120091009-1900
29. SVNKit 1.1.7 Implementation (Optional) 2.2.1.120091009-1900
30. SVNKit 1.2.2 Implementation (Optional) 2.2.1.120091009-1900
31. SVNKit 1.2.2 Implementation (Optional) 2.2.1.120091009-1900
32. SVNKit 1.3.0 Implementation (Optional) 2.2.1.120091009-1900
33. SVNKit 1.3.0 Implementation (Optional) 2.2.1.120091009-1900
34. Dali Java Persistence Tools 2.2.2.v200911250220-7L70APFBB0PSOTBgXV1je
35. Dali Java Persistence Tools - EclipseLink Support (Optional) 
2.2.2.v200911250220-65A9AkF77980FL7BB7
36. Data Tools Platform Connectivity 
1.7.2.v200909251450-71370ANEnlXi4gycmhPC6njFVDzOq
37. Data Tools Platform Open Data Access Designer 
1.7.2.v200909251450-7E7C7BDZRDIVDUKiEaXU
38. Data Tools Platform Open Data Access Runtime 
1.7.2.v200909251450-7H797BCcNBHBBHMWDbRK
39. Data Tools Platform Connectivity Documentation 
1.7.2.v200909251450-37D-7733L3D753L7BBF
40. Data Tools Platform Enablement 
1.7.2.v200909251450-7J8i7UBWwSLRWdU5OunZvF,PzOt
41. Data Tools Platform Enablement for Apache Derby 
1.7.2.v200909251450-7768dBmKDLfMjHuIFJW7SBI
42. Data Tools Platform Enablement for JDBC 
1.7.2.v200909251450-4-290B55W5P7G6RAH
43. Data Tools Platform Enablement for HSQLDB 
1.7.2.v200909251450-67B2AqGBJVKbalH3GP7SBI
44. Data Tools Platform Enablement for JDBC 
1.7.2.v200909251450-4-290B55W5P7G6RAH
45. Data Tools Platform Enablement for IBM 
1.7.2.v200909251450-7F47HFC75RTNSjXAYIRR
46. Data Tools Platform Enablement for Ingres 
1.7.2.v200909251450-540AkF77g7VBLBPH7
47. Data Tools Platform Enablement for JDBC 
1.7.2.v200909251450-4-290B55W5P7G6RAH
48. Data Tools Platform Enablement for Microsoft 
1.7.2.v200909251450-542AkF78Y7SBU9UAB
49. Data Tools Platform Enablement for MySQL 
1.7.2.v200909251450-546AkF78Y7R9PAX87
50. Data Tools Platform Enablement for ODA Designer
1.7.2.v200909251450 33285735335614153
51. Data Tools Platform Enablement for ODA Runtime 
1.7.2.v200909251450-7A7T78DZRDRID_KmFcNp
52. Data Tools Platform Enablement for Oracle 
1.7.2.v200909251450-548aAkF77g7XAO9aBB
53. Data Tools Platform Enablement for PostgreSQL 
1.7.2.v200909251450-542AkF77g7V9N9e77
54. Data Tools Platform Enablement for SAP MaxDB 
1.7.2.v200909251450-540AkF77g7V9N9e77
55. Data Tools Platform Enablement for Sqlite 
1.7.2.v200909251450-541AkF79P7N8NAQ97
56. Data Tools Platform Enablement for Sybase 
1.7.2.v200909251450-7E45F9NiNWv00XxPdVT
57. Data Tools Platform JDT Enablement 1.7.2.v200909251450-2-07w311A1A351453
58. Data Tools Platform Intro 1.7.2.v200909251450-26-7w311A17142212
59. Data Tools Platform Model Base 1.7.2.v200909251450-77-5CcNBCtCcCQJYZI
60. Data Tools Platform SQL Development Tools 
1.7.2.v200909251450-7N7q7GFDr_mX5M8JA538u8wWDRWU
61. Data Tools Platform SQL Development Tools Data Functions 
1.7.2.v200909251450-613BgnAiBPASF_KN
62. Data Tools Platform SQL Development Tools DDL Functionality 
1.7.2.v200909251450-261781302G85_ILGJxNiTuSn
63. Data Tools Platform SQL Development Tools DDL Generation Functionality 
1.7.2.v200909251450-7A-1F7RZHKFIwMhStRm
64. Data Tools Platform SQL Development Tools DDL Generation Functionality 
1.7.2.v200909251450-7A-1F7RZHKFIwMhStRm
65. Data Tools Platform SQL Development Tools Results View 
1.7.2.v200909251450-40290B56N517C9J5C
66. Data Tools Platform SQL Parsers 1.7.2.v200909251450-602BgJ99q9_90GXFJ
67. Data Tools Platform SQL Query Builder 1.7.2.v200909251450-79-0EVVFIxGhLaMvL_
68. Data Tools Platform SQL Schema Object Editor Feature 
1.7.2.v200909251450-42-8375LG5BJ93413
69. Data Tools Platform SQL Dev Tools Documentation 
1.7.2.v200909251450-37D-7733L3D753L7BBF
70. Data Tools Platform Tutorial Documentation 
1.7.2.v200909251450-26-311A16321A3557
71. Data Tools Platform User Documentation 
1.7.2.v200909251450-47C08w95ENAK6AFDFK7
72. EclEmma Java Code Coverage 1.4.3
73. Eclipse COS Client 1.1.101.R35x,20100125-7E79FGD9kKF67BPDZ7PKCE1911
74. Eclipse Faceted Project Framework 3.1.1.v200908111817-37718s733L4C58517F
75. Eclipse Faceted Project Framework JDT Enablement 
3.1.1.v200908111817-37718s733L56485K8H
76. Eclipse Java Development Tools 
3.5.2.r352,20100108-7r88FEwFIOWThoBlOiaGOtyhfZH6
77. Eclipse Java EE Developer Tools 
3.1.1.v200908101600-7_7EGrjFQRwRb4P511ebObS5XZhq
78. JST Enterprise Core 3.1.1.v200908101600-52FGEAkF7AG7CDLCFIE
79. JST Enterprise User Documentation 3.1.0.v200809051357-62FIXBgJ99nBPBjIHN_
80. EclipseLink JPA 1.1.3.v20091002,5404
81. EclipseLink MOXy 1.1.3.v20091002,5404
82. EclipseLink SDO 1.1.3.v20091002,5404
83. Eclipse Plug-in Development Environment 
3.5.2.R35x_v20100119-7Z7_FA2FDX-aXQYWqYDBz-zOqufo
84. Eclipse Web Developer Tools 3.1.1.v200908120400-7R77FStEVw2z07WtDz-OZrhL5C-3
85. Eclipse XML Editors and Tools 
3.1.1.v200907161031-7H6FM6DxtkM590eLGF98LRhdPKe0
86. WST Common UI 3.1.1.v200908102300-7B5FRHDhdMOGnoekn4gtEJxSmPxV
87. WST Common Core 3.1.1.v200908102300-71377FZ6F7RZHKDIwRIOwUN
88. WST XML Core 3.1.1.v200907161031-7C7OFPiF7RZHKFHgU1LkY6
89. WST XML User Documentation 3.1.1.v200906020459-50FR7AkF77f7MESDKEF
90. JavaScript Developer Tools 1.1.2.v200908101420-77-FGDCcNBDjBXM0BbFb
91. WST Common UI 3.1.1.v200908102300-7B5FRHDhdMOGnoekn4gtEJxSmPxV
92. WST Common Core 3.1.1.v200908102300-71377FZ6F7RZHKDIwRIOwUN
93. WST Server UI 3.1.1.v2009072713-7B3DPAUZRZxULQRALIM4256
94. WST Server Core 3.1.1.v2009072713-30Eu85734C3A5F
95. WST Server User Documentation 3.1.0.v200903071919-209L7w311Al2343145
96. WST Web Core 3.1.1.v200908120400-7H77FDCAT70GlivgHN1M2RkCBHz0
97. JavaScript Developer Tools 1.1.2.v200908101420-77-FGDCcNBDjBXM0BbFb
98. WST Common Core 3.1.1.v200908102300-71377FZ6F7RZHKDIwRIOwUN
99. WST Server Core 3.1.1.v2009072713-30Eu85734C3A5F
100.    WST Web Services Core 3.1.2.v200911111930-7L7RFTXFGtGcn6ij-pviG6462A
101.    WSDL 1.5 Feature 1.5.3.v200903071919 2407w311917141519
102.    WST XML Core 3.1.1.v200907161031-7C7OFPiF7RZHKFHgU1LkY6
103.    WST Web Services UI 3.1.2.v200911241834-7I6FPpEtEnGx_Csz-6wI8hoH_MRt
104.    Eclipse XML Editors and Tools 
3.1.1.v200907161031-7H6FM6DxtkM590eLGF98LRhdPKe0
105.    WST Common UI 3.1.1.v200908102300-7B5FRHDhdMOGnoekn4gtEJxSmPxV
106.    WST Common Core 3.1.1.v200908102300-71377FZ6F7RZHKDIwRIOwUN
107.    WST XML Core 3.1.1.v200907161031-7C7OFPiF7RZHKFHgU1LkY6
108.    WST XML User Documentation 3.1.1.v200906020459-50FR7AkF77f7MESDKEF
109.    WST Web Services Core 3.1.2.v200911111930-7L7RFTXFGtGcn6ij-pviG6462A
110.    WSDL 1.5 Feature 1.5.3.v200903071919 2407w311917141519
111.    WST Web Services User Documentation 3.1.0.v200903071919-44FGz90B55U5L80AFEK
112.    WST Web User Documentation 3.1.0.v200903071919-31DF85733J3E4D56BB
113.    Eclipse XSL Developer Tools 1.0.2.v200908271520-7R7T8sFIhIehWg454rq_Ul
114.    EMF - Eclipse Modeling Framework Runtime and Tools 2.5.0.v200906151043
115.    EMF Code Generation 2.5.0.v200906151043
116.    EMF Code Generation UI 2.4.0.v200906151043
117.    EMF Common 2.5.0.v200906151043
118.    EMF Common UI 2.5.0.v200906151043
119.    EMF Data Binding 1.1.0.v200906151043
120.    EMF - Eclipse Modeling Framework Core Runtime 2.5.0.v200906151043
121.    EMF Ecore Code Generator 2.5.0.v200906151043
122.    EMF Ecore Code Generator UI 2.5.0.v200906151043
123.    EMF Ecore Edit 2.5.0.v200906151043
124.    EMF Ecore Mapping 2.5.0.v200906151043
125.    EMF Ecore Mapping Editor 2.5.0.v200906151043
126.    EMF Edit 2.5.0.v200906151043
127.    EMF Edit Data Binding 1.1.0.v200906151043
128.    EMF Edit UI 2.5.0.v200906151043
129.    EMF Mapping 2.5.0.v200906151043
130.    EMF Mapping UI 2.5.0.v200906151043
131.    EMF Model Converter 2.5.0.v200906151043
132.    EMF Sample Ecore Editor 2.5.0.v200906151043
133.    FindBugs Feature 1.3.9.20090821
134.    Google App Engine Java SDK 1.6.1 1.6.1.v201201120043r35
135.    Google Plugin for Eclipse 3.5 2.5.1.v201201120043-rel-r35
136.    Google Web Toolkit SDK 2.4.0 2.4.0.v201201120043-rel-r35
137.    Graphical Editing Framework GEF 
3.5.1.v20090724-1146-77790819386696L3DA843563
138.    Graphical Editing Framework Draw2d 3.5.1.v20090724-1146-4407w311A192402452
139.    GWT Designer Core 2.5.0.r35x201112291019
140.    GWT Designer Editor 2.5.0.r35x201112291029
141.    GWT Designer GPE 2.5.0.r35x201201030222
142.    Java EE IDE Feature 1.2.1.20090918-0703
143.    Eclipse Platform 3.5.1.R35x_v20090910-9gEeGl_FthkNDSP20dXdTha0u9GFDPn83DGB7
144.    Eclipse Help System 1.1.1.R35x_v20090811-7e7eFAnFEx2XZoYzOuPgIfwD
145.    Eclipse RCP 3.5.1.R35x,20090811-9SA0FxVFqE700L1ARMrfc06e7BA6
146.    Equinox p2 Provisioning 1.1.1.R35x,20090811-7u6FbEFYXklqWdbSOgnpRg2932
147.    JD -Eclipse Plug-in 0.1.3
148.    JST Server Adapters 3.1.0.v20090311c-20-7w311A1422c
149.    JST Server Adapters 3.1.1.v20090813-777HFDPCcNBBzOBjJU4BA8
150.    JST Server UI 3.1.1.v20090817-7A2F13139xFaH0vXGLUCTMQ2355
151.    JST Server Core 3.1.2.v20100127-31Ev85734C3C6H
152.    JST Server User Documentation 3.1.0.v200903071919-2-DF7w311916261244
153.    JST Web UI 3.1.1.v200908121609-7E77FBiD1wYcICNdz-5z-9PGqZCy
154.    JST Web Core 3.1.1.v200908121609-757CG-dF1hIeq7kV6qxaLD
155.    JST Web User Documentation 3.1.0.v200809240655-20A77w311A17263135
156.    Metrics plugin for Eclipse 1.3.6
157.    Mylyn Connector: Sonar (Optional) 2.3.0.20111209-1529
158.    Mylyn Context Connector: Eclipse IDE 3.6.4.v20111118-0100
159.    Mylyn Context Connector: Java Development 3.6.4.v20111118-0100
160.    Mylyn Context Connector: Team Support 3.6.4.v20111118-0100
161.    Mylyn Task Focused Interface 3.6.4.v20111118-0100
162.    Mylyn Task List 3.6.4.v20111118-0100
163.    Mylyn Tasks Connector: Bugzilla 3.6.4.v20111118-0100
164.    Mylyn WikiText 1.5.4.v20111118-0100
165.    PMD Plug-in 3.2.6.v200903300643
166.    Remote System Explorer End User Runtime 
3.1.2.R31x,200911051655-7L5AF8wqa1PUdt0M8wtNiNhORez 
167.    RSE Core 3.1.1.v200907301400-7a7CEwFC7sRTNSwodfvu
168.    RSE DStore Services 3.1.1.v200907301400-7L6E5EVVFGMG4Ty0VFU
169.    RSE FTP Service 3.0.101.v200909160005-780A890B55W50AP9A91
170.    RSE Local Services 2.1.101.v200907301400-7B4EqBgJ99r9dJ_GM9E
171.    RSE SSH Services 3.0.1.v200909160005-7A2BhAkF77g7VEYDG7C
172.    RSE Telnet Service 2.2.1.v200908051630-7730AkF77g7UDXELEF
173.    RSE Terminals UI 1.0.2.R31x,200911051655-77-49uC8Ez-BY3y4Q1E224382252
174.    Target Management Terminal Widget 
3.0.2.R31x,200911051655-4-17542yaw312113271141
175.    Remote System Explorer User Actions 
1.1.100.v200905272300-31A785733L3D7H7933
176.    Sonar Integration for Eclipse (Required) 2.3.0.20111209-1529
177.    Spring IDE Core (required) 2.2.8.200911091054 RELEASE
178.    Target Management Terminal SSH Connector 
2.0.100.v200905272300-3-8Z7w311A16383411
179.    Target Management Terminal Telnet Connector 
2.0.101.v200907301400-3-8_7w311A1A271116
180.    Target Management Terminal View 2.0.101.v200907301400-3-8_7w311A18412511
181.    Usage Data Collector 1.1.1.R201001291118
182.    Web Page Editor (Optional) 2.2.2.v201001271707-437990B5855B7A2242
183.    WindowBuilder Core 1.2.0.r35x201112290923
184.    WindowBuilder CSS Support 1.2.0.r35x201112290951
185.    WindowBuilder XML Core (requires Eclipse WTP/WST) 1.2.0.r35x201112291009
186.    WST Server Adapters 3.1.0.v200906020451-51DFAkF77g7GCNc
187.    D: \ Eclipse \ AOSI-EclipseD2I-v1.5.2 \ dropins \ eclipse-clearcase
188.    Rational ClearCase Remote Client 7.0.1.D101208
189.    Rational ClearCase Remote Client pour Eclipse 7.0.1.D101208
190.    Fonction d'importation de Rational ClearCase Remote Client pour Eclipse 
7.0.0.200706120
191.    Fonction d'importation NL1 de Rational ClearCase Remote Client pour 
Eclipse 7.0.0.20061020A
192.    Rational ClearCase Visual Studio Integration 7.0.1.D101208

```

Original comment by `weremarc` on 24 Jan 2012 at 9:03

**Comment by GoogleCodeExporter** [comment]
```
I forgot to mention that I am on Eclipse 3.5.1.R35
```

Original comment by `weremarc` on 24 Jan 2012 at 9:20

**Comment by GoogleCodeExporter** [comment]
```
Thanks a lot for reporting.
Can you do the following. 
1) Create a new workspace.
2) Reproduce the error.
3) send the error log in your_workspace_directory/.metadata/.log
```

Original comment by `chintanr...@google.com` on 24 Jan 2012 at 5:22

**Comment by GoogleCodeExporter** [comment]
```
- I created a workspace "workspace_test"*
- I created a new project "test_main"
- I created a new class Main in the default package with the following code:

The error appeared when I was asking completion for 
System.console().printf(arg0, arg1)



The complete code written :


public class Main {

    /**
     * @param args
     */
    public static void main(String[] args) {
        // TODO Auto-generated method stub
        System.console().printf(arg0, arg1)

    }

}



```

Original comment by `weremarc` on 25 Jan 2012 at 9:12

Attachments:
- [.log](https://storage.googleapis.com/google-code-attachments/google-plugin-for-eclipse/issue-49/comment-5/.log)

**Comment by GoogleCodeExporter** [comment]
```
We are working on this.
```

Original comment by `chintanr...@google.com` on 27 Feb 2012 at 10:30

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 2.5.2. Please check.
```

Original comment by `chintanr...@google.com` on 7 Mar 2012 at 12:32
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Appears to be fixed for me. Thank you, this was driving me nuts!
```

Original comment by `fay...@gmail.com` on 7 Mar 2012 at 4:00
